ci/check-compiles.sh

```diff
@@ -6,3 +6,4 @@ cargo check
 cargo doc
 cargo doc --document-private-items
 cd fuzz && cargo check --features=stdin_fuzz
+cd ../lightning && cargo check --no-default-features --features=no_std
```

Comment:
In `.github/workflows/build.yml` can you also add a `cd lightning && cargo test --verbose --color always --features hashbrown` to run the test suite with hashbrown + std?

Reply:
Which builds should that run under, or do you want a new build target?

Reply:
Uhhmmmmm, yes? I don't think it matters too much, maybe in the existing ones is simplest, given it's ~one line?